verilog/analysis/symbol_table_test.cc

```diff
@@ -6132,6 +6132,79 @@ TEST(BuildSymbolTableTest, AnonymousStructTypeNestedMemberReference) {
   }
 }
 
+TEST(BuildSymbolTableTest, AnonymousEnumTypeData) {
+  TestVerilogSourceFile src("enumy.sv",
+                            "enum {\n"
+                            "  idle, busy\n"
+                            "} data;\n");
+  const auto status = src.Parse();
+  ASSERT_TRUE(status.ok()) << status.message();
+  SymbolTable symbol_table(nullptr);
+  const SymbolTableNode& root_symbol(symbol_table.Root());
+
+  const auto build_diagnostics = BuildSymbolTable(src, &symbol_table);
+  EXPECT_EMPTY_STATUSES(build_diagnostics);
+
+  // Expect one anonymous enum definition and reference.
+  EXPECT_EQ(root_symbol.Value().anonymous_scope_names.size(), 1);
+  ASSERT_EQ(root_symbol.Children().size(), 2);
+  // Find the symbol that is a enum (anon), which is not "data".
+  const auto found =
+      std::find_if(root_symbol.Children().begin(), root_symbol.Children().end(),
+                   [](const SymbolTableNode::key_value_type& p) {
+                     return p.first != "data";
+                   });
+  ASSERT_NE(found, root_symbol.Children().end());
+  const SymbolTableNode& anon_enum(found->second);
+  const SymbolInfo& anon_enum_info(anon_enum.Value());
+  EXPECT_EQ(anon_enum_info.metatype, SymbolMetaType::kEnum);
+  EXPECT_TRUE(anon_enum_info.local_references_to_bind.empty());
+
+  // Enum has two members.
+  MUST_ASSIGN_LOOKUP_SYMBOL(idle, anon_enum, "idle");
+  EXPECT_EQ(idle_info.metatype, SymbolMetaType::kUnspecified);
+  EXPECT_EQ(idle_info.file_origin, &src);
+  //EXPECT_EQ(
```

Comment:
@fangism And here?

Reply:
I wonder if the enum values should have user-defined type pointing to the anonymous enclosing enum type, like values of an `enum class` in C++.  We need a bit of language expertise here.